src/Umbraco.Web/Editors/ParameterSwapControllerActionSelector.cs

```diff
@@ -25,33 +35,92 @@ public ParameterSwapControllerActionSelector(params ParameterSwapInfo[] actions)
         {
             _actions = actions;
         }
+
         public override HttpActionDescriptor SelectAction(HttpControllerContext controllerContext)
         {
             var found = _actions.FirstOrDefault(x => controllerContext.Request.RequestUri.GetLeftPart(UriPartial.Path).InvariantEndsWith(x.ActionName));
 
             if (found != null)
             {
-                var id = HttpUtility.ParseQueryString(controllerContext.Request.RequestUri.Query).Get(found.ParamName);
+                HttpActionDescriptor method;
+                if (TryBindFromUri(controllerContext, found, out method))
+                {
+                    return method;
+                }
 
-                if (id != null)
+                //if it's a post we can try to read from the body and bind from th
```

Comment:
Tiny typo, I know what you mean as been reading this PR but out of context someone else may not know you mean about binding to the POST'd JSON value